### DISCUSSION TITLE
Could you run CodeHike in ReactMarkdown in the client?

### DISCUSSION THREAD

**Discussion by MildTomato** [code-hike/codehike#323] [Ideas and feedback]
Thinking could we use codehike in the client like this, using ReactMarkdown (or something similar)

```js
<ReactMarkdown
  remarkPlugins={[
    [remarkCodeHike, { autoImport: false, theme: codeHikeTheme }],
  ]}
  components={components}
>
 {markdown}
</ReactMarkdown>;
```

just get an fs error with this
<img width="943" alt="Screenshot 2023-02-10 at 4 04 02 AM" src="https://user-images.githubusercontent.com/8291514/217924792-48537742-7d16-43e5-a1be-3e8c35f6678a.png">

**Comment by pomber** [comment]
It works on the client (see [the playground](https://github.com/code-hike/codehike/tree/next/playground)) but not with Next.js right now (because of Next.js webpack config). 

I think it should work after I fix https://github.com/code-hike/lighter/issues/6. 

By the way, it's usually a bad idea to run it on the client, except for things like playgrounds or experiments.

**Comment by MildTomato** [comment]
> By the way, it's usually a bad idea to run it on the client, except for things like playgrounds or experiments.

OK then, this might be a bad idea for this [type of page](https://supabase.com/docs/reference/javascript/installing). All of the code blocks on the right would be using codehike

**Reply by pomber** [reply]
because of the tabs for each endpoint? I think you can do that at build time. 

You can use Spotlight (see [this example](https://github.com/code-hike/codehike/issues/198#issuecomment-1160221396)), 

Or if you want more flexibility you can make your own EndpointTabs component and pass the code as children in mdx.

````mdx

<EndpointTabs>

<EndpointTab title="Getting your data">

```js
const { data, error } = await supabase
  .from('countries')
  .select()
```

```
...
```

</EndpointTab>

...

</EndpointTabs>


````

**Comment by RubanChristian** [comment]


**Reply by MildTomato** [reply]
This isn't using `<ReactMarkdown>` though ? 

The page isn't a markdown page, and we're using `next-mdx-remote` 😬
